server/tests/user.integration.spec.js

```diff
@@ -45,51 +56,88 @@ describe('Notifications API:', () => {
             .then(setTimeout(done, 1000));
     });
 
-    describe('Create User and Update Device Token', () => {
+    describe('Require authorization', () => {
         it('Should fail without Authorization header', () => request(app)
             .post(`${baseURL}/users`)
-            .send(testUserObject)
+            .send(userA)
             .expect(httpStatus.UNAUTHORIZED)
         );
 
         it('Should fail with badtoken in Authorization header', () => request(app)
             .post(`${baseURL}/users`)
             .set('Authorization', 'Bearer badtoken')
-            .send(testUserObject)
+            .send(userA)
             .expect(httpStatus.UNAUTHORIZED)
         );
+    });
+
+    describe('Create User and Update Device Token', () => {
+        it('should create and respond with a user', () => request(app)
+            .post(`${baseURL}/users`)
+            .set('Authorization', `Bearer ${auth}`)
+            .send(userA)
+            .expect(httpStatus.OK)
+            // eslint-disable-next-line no-underscore-dangle
+            .then(res => User.findById(res.body.user._id))
+            .then((user) => {
+                // eslint-disable-next-line no-unused-expressions
+                expect(user).to.not.be.null;
+                expect(user.username).to.equal(userA.username);
+                expect(user.uuid).to.equal(userA.uuid);
+            })
+        );
 
-        it('should should create and respond with a user', () => request(app)
+        it('should return success but not create a duplicate user', () => request(app)
             .post(`${baseURL}/users`)
             .set('Authorization', `Bearer ${auth}`)
-            .send(testUserObject)
+            .send(userA)
```

Comment:
If we `POST` to `/users` with a user that already exists, shouldn't it respond with a 409 or something?

Reply:
That would change the API so I think it's out of scope for this already large PR.